### ISSUE TITLE
feature request: forward-zones with forward and continue behaviour while recursing

### ISSUE THREAD

**Issue by Habbie** [PowerDNS/pdns#590]
Currently entries in forward-zones or forward-zones-file are acting as forward only.

It might be a good idea to have them configurable as forward and continue.

For example:

On PDNS recursor R:
forward-zones=22.172.in-addr.arpa,$IP_DNS_A

On PDNS authoritative A in zone 23.172.in-addr.arpa:
42 NS $DNS_B

On PDNS authoritative B exists zone 43.23.172.in-addr.arpa.

Currently R responds only with NOERROR, delivering NS records, the request to A is just forwarded and the answer handed back to the client without further processing the NS records.

To clarify things, the relevant parts of the IRC-Log:

```
15:13:40 < abraxxl> I have a recursor and two DNS servers. The recursor has an entry in forwarders for +foo.bar, DNS server A is authoritative for. In foo.bar are existing NS recor
                    recurse because of the prefix "+" before foo.bar in forward zones file. Is my guess correct?
15:56:37 < abraxxl> forget my question it is nonsense, the recursion-desired bit is ignored by an authoritative server
15:57:26 < ZaphodB> abraxxl: with the + prefix the query for baz.foo.bar should be forwarded to the server you specified in the forward-zones-file but with the recursion desired bi
15:59:44 < ZaphodB> if you want to just handle this one domain by auth servers locally the 'regular' forwarding that does not set the recursion desired bit is for you need to make 
                    then this should become a regular recursive query again i think.
16:18:22 < abraxxl> ZaphodB: yes, I expect the recursor to ask DNS server A for baz.foo.bar: DNS A should answer with NS records and the recursor should ask DNS server B according 
16:53:04 < Zugschlus> So, the real question of abraxxl is: When a recursor R asks an auth server A via forward.zones, and A answers with an referral to B, is the recursor supposed 
                      resolving process?
16:53:44 < ahu> forward zones 'only' injects a hard NS record for that exact domain
16:53:56 < ahu> however
16:54:00 < Zugschlus> so the recursor should recurse to B?
16:54:02 < ahu> it assumes that the answer it gets is 'done'
16:54:07 < ahu> no, it is not quite an NS record
16:54:08 < ahu> sorry
16:54:20 < ahu> it is a magic 'hard' NS record that says 'ask them and relay back the answer'
16:54:42 < Zugschlus> so the recursor will return the referral to the client which is of course not what the client expects?
16:54:44 < ahu> it does not then continue processing
16:54:59 < ahu> well, for the people we made this for, this solved their problem
16:55:07 < ahu> they wanted 'forward'
16:55:14 < ahu> they did not want 'insert NS records'
16:55:22 < ahu> so the use case is different
16:55:23 < Zugschlus> abraxxl seems to want "forward and continue"
16:55:37 < ahu> he wants a persistent stub zone I think
16:55:45 < Zugschlus> ouch, that's bind lingo
16:55:50 < ahu> yes
16:55:51 < ahu> ;-)
16:55:55 < Zugschlus> need to twist my brain to remember what that means
16:56:10 < ahu> I think it means what I said, 'insert a persistent NS record'
16:57:08 < Zugschlus> hm
16:57:23 < Zugschlus> So our recommendation to abraxxl would be: Insert more forward.zones entries
16:57:38 < Zugschlus> (and watch for debris if the people running the auth server change things)
16:57:58 < ahu> yeah, and author a really well written ticket
16:58:03 < ahu> with a feature request
```

**Comment by pieterlexis** [comment]
I _think_ this was fixed with #3941

**Comment by pieterlexis** [comment]
Marking this fixed
